sdcm/utils/common.py

```diff
@@ -98,12 +108,24 @@ def inner(*args, **kwargs):
                 except self.allowed_exceptions as ex:
                     LOGGER.debug("'%s': failed with '%r', retrying [#%s]", func.__name__, ex, i)
                     time.sleep(self.sleep_time)
-                    if i == self.n - 1:
-                        LOGGER.error("'%s': Number of retries exceeded!", func.__name__)
+                    if i == self.n - 1 or (end_time and time.time() > end_time):
+                        LOGGER.error(f"'{func.__name__}': Number of retries exceeded!")
                         raise
+
         return inner
 
 
+class timeout(retrying):  # pylint: disable=invalid-name,too-few-public-methods
```

Comment:
I hope this is the last comment on this one

Let's move it to the same file retrying sits ?

Reply:
my mistake, It was already in place.
for some reason I thought it was in sct_events.py

sorry for the confusion